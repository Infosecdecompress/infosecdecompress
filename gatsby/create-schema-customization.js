const createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions;
  const typeDefs = `
    type MarkdownRemark implements Node {
      frontmatter: Frontmatter
    }
    type SocialImage {
      publicURL: String
    }
    type Frontmatter {
      socialImage: SocialImage
    }
  `;

  createTypes(typeDefs);
};

module.exports = createSchemaCustomization;