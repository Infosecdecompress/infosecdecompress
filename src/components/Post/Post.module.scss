@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .comments {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .button {
    border: 1px solid $color-gray-border;
    border-radius: $button-border-radius;
    color: $color-base;
    display: block;
    font-size: $typographic-base-font-size;
    font-weight: normal;
    height: $button-height;
    line-height: $button-height;
    margin-left: auto;
    margin-right: auto;
    max-width: 90px;
    text-align: center;
    @include padding(0, 1);
    @include margin-top(1);

    &:hover,
    &:focus {
      color: $color-primary;
    }
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }

    .comments {
      @include padding-equal(0);
    }

    .button {
      left: 30px;
      max-width: none;
      position: fixed;
      top: 30px;
      @include margin-equal(0);
    }
  }
}
