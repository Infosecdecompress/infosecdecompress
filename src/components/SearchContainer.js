import React, { Component } from "react"
import Axios from "axios"
import * as JsSearch from "js-search"

class Search extends Component {
  state = {
    itemList: [],
    search: [],
    searchResults: [],
    isLoading: true,
    isError: false,
    searchQuery: "",
  }
  /**
   * React lifecycle method to fetch the data
   */
  async componentDidMount() {
    Axios.get("/feed-1.json")
      .then(result => {
        const feedData = result.data
        this.setState({ itemList: feedData.items })
        this.rebuildIndex()
      })
      .catch(err => {
        this.setState({ isError: true })
        console.log("====================================")
        console.log(`Something bad happened while fetching the data\n${err}`)
        console.log("====================================")
      })
  }

  /**
   * rebuilds the overall index based on the options
   */
  rebuildIndex = () => {
    var REGEX = /\s+/; // Split on spaces
    const tokenizer = {
      tokenize(text) {
        return text
          .split(REGEX)
          .filter(
            (text) => text // Filter empty tokens
          );
      }
    }
    const { itemList } = this.state
    const dataToSearch = new JsSearch.Search("id")
    dataToSearch.tokenizer = tokenizer
    /**PrefixIndexStrategy   AllSubstringsIndexStrategy ExactWordIndexStrategy  */
    dataToSearch.indexStrategy = new JsSearch.PrefixIndexStrategy()
    dataToSearch.sanitizer = new JsSearch.LowerCaseSanitizer()
    /**defines the search index  https://github.com/bvaughn/js-search#configuring-the-search-index*/
    dataToSearch.searchIndex = new JsSearch.TfIdfSearchIndex("title")
    dataToSearch.addIndex("title") // sets the index attribute for the data
    dataToSearch.addIndex("content") // sets the index attribute for the data
    dataToSearch.addDocuments(itemList) // adds the data to be searched
    this.setState({ search: dataToSearch, isLoading: false })
  }

  /**
   * handles the input change and perform a search with js-search
   * in which the results will be added to the state
   */
  searchData = e => {
    const { search } = this.state
    const queryResult = search.search(e.target.value)
    this.setState({ searchQuery: e.target.value, searchResults: queryResult })
    var status = document.getElementById("result");
    if (e.target.value != "") {
      status.style.display = "block";
    }
    else {
      status.style.display = "none";
    }
  }
  handleSubmit = e => {
    e.preventDefault()
  }

  render() {
    const { itemList, searchResults, searchQuery } = this.state
    const queryResults = searchQuery === "" ? itemList : searchResults
    return (
      <div style={{margin:"5px auto"}}>
        <div style={{ margin: "0 auto" }}>
          <form onSubmit={this.handleSubmit}>
            <div style={{ margin: "0 auto" }}>
              <input
                id="Search"
                value={searchQuery}
                onChange={(evt) => this.searchData(evt)}
                placeholder="搜尋文章"
                style={{ 
                  margin: "0 auto", 
                  width: "250px",
                  padding: "5px 5px",
                  display: "inline-block",
                  border: "1px solid #ccc",
                  borderRadius: "4px",
                  boxSizing: "border-box",
                }}
              />
            </div>
          </form>
          <div>
            <table id="result"
              style={{
                width: "250px",
                borderCollapse: "collapse",
                borderRadius: "4px",
                border: "1px solid #ccc",
                display: "none",
              }}
            >
              <tbody>
                {queryResults.map(item => {
                  return (
                    <tr key={`row_${item.id}`}>
                      <td
                        style={{
                          fontSize: "14px",
                        }}
                      >
                        <a href={item.url}>
                          {item.title}
                        </a>
                      </td>
                    </tr>
                  )
                })}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    )
  }
}
export default Search