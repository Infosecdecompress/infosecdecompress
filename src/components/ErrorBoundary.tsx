import React from "react"

import { ErrorBoundary } from "@sentry/react"
import { Link } from "gatsby"

const Fallback = ({ error, resetError }) => (
  <div role="alert">
    <p>Something went wrong</p>
    <pre>{error.message}</pre>
    <Link to="/" onClick={resetError}><a>Try again</a></Link>
  </div>
)
const ErrorBoundaryContainer = ({ children }) => <ErrorBoundary fallback={Fallback}>{children}</ErrorBoundary>

export default ErrorBoundaryContainer