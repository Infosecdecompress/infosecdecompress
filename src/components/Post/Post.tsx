import React from "react";

import { Link } from "gatsby";

import type { Node } from "@/types";

import { Author } from "./Author";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Meta } from "./Meta";
import { Tags } from "./Tags";

import * as styles from "./Post.module.scss";

type Props = {
  post: Node;
};

const Post: React.FC<Props> = ({ post }) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles.post}>
      <Link className={styles.button} to="/">
        文章列表
      </Link>

      <div className={styles.content}>
        <Content body={html} title={title} />
      </div>

      <div className={styles.footer}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles.comments}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
