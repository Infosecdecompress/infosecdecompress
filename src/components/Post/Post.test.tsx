import React from "react";
import renderer from "react-test-renderer";

import { StaticQuery, useStaticQuery } from "gatsby";

import { Post } from "@/components/Post";
import * as mocks from "@/mocks";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("Post", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  it("renders correctly", () => {
    const props = { post: mocks.markdownRemark };
    const tree = renderer.create(<Post {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
