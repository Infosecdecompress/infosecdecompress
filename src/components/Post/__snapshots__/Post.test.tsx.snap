// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<div>
  <div>
    <a
      className=""
      href="/"
    >
      文章列表
    </a>
  </div>
  <div>
    <div>
      <h1>
        Perfecting the Art of Perfection
      </h1>
      <div
        dangerouslySetInnerHTML={
          {
            "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
          }
        }
      />
    </div>
  </div>
  <div>
    <div>
      <p>
        發佈於
         
        2016年8月31日
      </p>
    </div>
    <div>
      <center>
        <p>
          資安解壓縮網站內容除特別註記外皆以
          <a
            href="http://creativecommons.org/licenses/by-sa/4.0/"
            rel="noopener noreferrer"
          >
            CC BY-SA 4.0
          </a>
          方式授權
          <br />
          對於這個節目有什麼想法，或想要聽什麼內容都歡迎到
          <a
            href="/pages/contacts"
            rel="noopener noreferrer"
          >
            聯絡資訊
          </a>
          告訴我們
        </p>
      </center>
    </div>
  </div>
  <div />
</div>
`;
